### PR TITLE
Add <li> numbering tests for weird cases

### PR DESCRIPTION
**Pull request by domenic**
See https://github.com/whatwg/html/pull/2002

### CODE REVIEW

File: html/semantics/grouping-content/the-li-element/grouping-li-no-list-owner-ref.html [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+<title>If no ancestors generate CSS boxes, the list item has no owner, and thus gets numbered as 0</title>
+<link rel="author" title="Domenic Denicola" href="mailto:d@domenic.me">
+<link rel="help" href="https://html.spec.whatwg.org/multipage/semantics.html#ordinal-value">
+<link rel="help" href="https://html.spec.whatwg.org/multipage/semantics.html#list-owner">
+
+<link rel="match" href="grouping-li-reftest-no-list-owner-ref.html">
+
+<style>
+  html, body {
+    display: contents;
```
Comment (zcorpan) [inline html/semantics/grouping-content/the-li-element/grouping-li-no-list-owner-ref.html]: Don't use display: contents in the ref

File: html/semantics/grouping-content/the-li-element/grouping-li-reftest-display-list-item-ref.html [hunk 2]
```diff
@@ -0,0 +1,40 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>display: list-item on non-&lt;li> elemnets</title>
```
Comment (zcorpan) [inline html/semantics/grouping-content/the-li-element/grouping-li-reftest-display-list-item-ref.html]: typo elemnets

File: html/semantics/grouping-content/the-li-element/grouping-li-reftest-display-list-item.html [hunk 3]
```diff
@@ -0,0 +1,44 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>display: list-item on non-&lt;li> elemnets</title>
```
Comment (zcorpan) [inline html/semantics/grouping-content/the-li-element/grouping-li-reftest-display-list-item.html]: typo elemnets

**Review by zcorpan — APPROVED** [review]
LGTM % nits

File: html/semantics/grouping-content/the-li-element/grouping-li-no-list-owner-ref.html (hunk 1 above)
Comment (domenic) [inline html/semantics/grouping-content/the-li-element/grouping-li-no-list-owner-ref.html]: It has an impact on padding/margin/etc. It seemed better to use this here than to tweak those.

**Review by zcorpan — APPROVED** [review]

**Pull request merged**